src/plugins/circular_view/src/CircularItems.cpp

```diff
@@ -436,14 +436,15 @@ bool CircularAnnotationLabel::tryPushClockwise() {
     const int engagedIndex = ra->engagedLabelPositionToLabel.key(this, -1);
     SAFE_POINT(-1 != engagedIndex, "Unexpected label position index", false);
     const int numberOfEngagedIndex = suitableLabelPositionIndexes.indexOf(engagedIndex);
-    SAFE_POINT(-1 != numberOfEngagedIndex, "Unexpected label position index", false);
 
-    if ((suitableLabelPositionIndexes.size() - 1) == numberOfEngagedIndex) {
+    if ((suitableLabelPositionIndexes.size() - 1) == numberOfEngagedIndex
+            || -1 == numberOfEngagedIndex) {
```

Comment:
Do not use yoda style and add a comment why the index should be skipped as well as no index at all, so it will be easy to understand to anyone who reads this code in the future.